expression/expression_test.go

```diff
@@ -225,6 +281,58 @@ func TestGetExpressionValue(t *testing.T) {
 	})
 }
 
+func TestExpressionValidate(t *testing.T) {
+	warnValue := float64(60)
+	errorValue := float64(90)
+	Convey("Default Expressions", t, func() {
+		expressions := []string{
+			"t1 >= ERROR_VALUE ? ERROR : (t1 >= WARN_VALUE ? WARN : OK)",
+			"t1 <= ERROR_VALUE ? ERROR : (t1 <= WARN_VALUE ? WARN : OK)",
+			"t1 >= WARN_VALUE ? WARN : OK",
+			"t1 >= ERROR_VALUE ? ERROR : OK",
+			"t1 <= WARN_VALUE ? WARN : OK",
+			"t1 <= ERROR_VALUE ? ERROR : OK",
+			"WARN",
+		}
+		for _, expr := range expressions {
+			triggerExpression := TriggerExpression{
+				Expression:  ptr(expr),
+				WarnValue:   &warnValue,
+				ErrorValue:  &errorValue,
+				TriggerType: moira.ExpressionTrigger,
+			}
+			err := triggerExpression.Validate()
+			So(err, ShouldBeNil)
+		}
+	})
+	Convey("Invalids", t, func() {
+		expressions := []string{
+			"t1 > 10 ? ok : (t2 * 10 : ddd ? warn)",
+			"t1 < WARN_VALUE ? ok ?",
+			"t1 > 10 ? ok : (t2 * 10 : error ? warn)",
+		}
+		for _, expr := range expressions {
+			triggerExpression := TriggerExpression{
+				Expression:  ptr(expr),
+				WarnValue:   &warnValue,
+				ErrorValue:  &errorValue,
+				TriggerType: moira.ExpressionTrigger,
+				AdditionalTargetsValues: map[string]float64{
+					"t2": 2,
+					"t3": 3,
+				},
+			}
+			err := triggerExpression.Validate()
+			t.Log(err)
+			So(err, ShouldNotBeNil)
+		}
+	})
+}
+
+func ptr[T any](obj T) *T {
```

Comment:
So, why do you need this function if all your expressions are strings? I mean of the same type

Reply:
there were some tests that had pointers to float64 like `WarnValue` and `ErrorValue`. I think I didn't later use them, can I change it to `stringPtr`?

Reply:
What I mean is that `ptr` in the code is only used for expressions that are always strings, you could just do as in `WarnValue` or `ErrorValue` and write:
```go
Expression: &expr,
```
Is not it?

Reply:
Ohh true, the function is truly useless here